frame/staking/src/pallet/mod.rs

```diff
@@ -1243,18 +1241,21 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::set_controller())]
 		pub fn set_controller(origin: OriginFor<T>) -> DispatchResult {
 			let stash = ensure_signed(origin)?;
-			let old_controller = Self::bonded(&stash).ok_or(Error::<T>::NotStash)?;
 
-			if <Ledger<T>>::contains_key(&stash) {
```

Comment:
nit: This was cheaper fail early precondition.

Reply:
Yes, true. But since this is just temporary code logic, I think it's OK as is.